### PR TITLE
[Feature Request] - Optimizing Image cropper and reduce image while uploading

### PR DESCRIPTION
**Pull request by MAXASSASSINS**
## [Feature Request] - Optimize Image cropper and reduce image while uploading

Closes: #496

## Description of Changes
- Move all the functionality from Mentor Form Registration to Person Details Component.
- Added image cropper to Person Details component.
- Added Cloudinary upload to Person Details component
-  Fix the path of CSS file of slider component

## Checklist:

- [x] My code adheres to the established style guidelines of this project.
- [x] I have conducted a self-review of my code.
- [x] I have included comments in areas that may be difficult to understand.
- [x] I have made corresponding updates to the project documentation.
- [x] My changes have not introduced any new warnings.

## Screenshots

https://github.com/anmode/grabtern-frontend/assets/97503316/3ddf42ed-1139-44c8-81c2-fecadd3da763

https://github.com/anmode/grabtern-frontend/assets/97503316/41eb78da-0845-4f89-9ff7-302c7b89bedf

### CODE REVIEW

File: components/MentorRegistration/components/PersonDetails.js [hunk 1]
```diff
@@ -59,8 +61,64 @@ function PersonDetails({
...
+
+      fileReader.onload = () => {
+        resolve(fileReader.result);
+      };
+
+      fileReader.onerror = (error) => {
+        reject(error);
+      };
+    });
+  };
```
Comment (anmode) [inline components/MentorRegistration/components/PersonDetails.js:81]: sorry we are not using base64 anymore...

**Review by anmode — CHANGES_REQUESTED** [review]
Please update your branch. I see new changes haven't come

**Comment by MAXASSASSINS** [comment]
> Please update your branch. I see new changes haven't come

But on my side it shows that my branch is already ahead of anmod:develop branch

Screenshot 2023-07-13 at 9 32 21 PM

File: components/MentorRegistration/components/PersonDetails.js (hunk 1 above)
Comment (MAXASSASSINS) [reply components/MentorRegistration/components/PersonDetails.js:81]: This is for the image cropper component. Mentor image will be storing the cloudinary url.

Comment (anmode) [reply components/MentorRegistration/components/PersonDetails.js:81]: oh okayyy!!

**Comment by anmode** [comment]
I think it is cool, I'll review it tomorrow.. Probably no more changes required will merge tomorrow..
if you want you can work on another issues/

File: components/MentorRegistration/components/PersonDetails.js [hunk 2]
```diff
@@ -59,8 +61,64 @@ function PersonDetails({
...
+  const handleImageCropperChange = async (imageSrc) => {
+    setShowImageCropper(false);
+    if (!imageSrc) return;
+    setImageSrc(imageSrc);
+    const imageClouindaryUrl = await uploadToCloudinary(imageSrc);
+    handleUploadImageChange(fileName, imageClouindaryUrl);
+  };
+
+  const uploadToCloudinary = async (imageSrc) => {
+    const blob = await fetch(imageSrc).then((r) => r.blob());
```
Comment (glokta1) [inline components/MentorRegistration/components/PersonDetails.js:101]: Don't mix `await` and `then` inside a function. It leads to buggy and hard-to-read code. Please use `await` here.

**Review by glokta1 — CHANGES_REQUESTED** [review]

File: components/MentorRegistration/components/PersonDetails.js (hunk 2 above)
Comment (MAXASSASSINS) [reply components/MentorRegistration/components/PersonDetails.js]: Hey @glokta1 i have made the changes

**Review by glokta1 — APPROVED** [review]

**Comment by glokta1** [comment]
LGTM. Great work! @MAXASSASSINS

**Pull request merged**